### DISCUSSION TITLE
Why Magick::Blob::base64() receives argument base64_ by copy?

### DISCUSSION THREAD

**Discussion by eao197** [ImageMagick/ImageMagick#6708] [Contribute]
Method Magick::Block::base64() has the following prototype:
```c++
void Magick::Blob::base64(const std::string base64_);
```
It means that `base64_` will be passed by a copy in the following example:
```c++
Magick::Blob blob;
std::string base64_data = read_image_data_from_somewhere();
blob.base64(base64_data); // A copy of base64_data will be passed to base64() here!
```
It seems like a typo in method prototype: `const std::string base64_` instead of `const std::string & base64_`.

**Comment by dlemstra** [comment]
This change will require us to do `.so` bump because we break the ABI. We will try to make this change when we need to make other ABI/API changes.
